pkg/controller/volume/persistentvolume/index.go

```diff
@@ -211,11 +211,18 @@ func findMatchingVolume(
 		}
 
 		// filter out:
+		// - volumes in non-available phase
 		// - volumes bound to another claim
 		// - volumes whose labels don't match the claim's selector, if specified
 		// - volumes in Class that is not requested
 		// - volumes whose NodeAffinity does not match the node
-		if volume.Spec.ClaimRef != nil {
+		if volume.Status.Phase != v1.VolumeAvailable {
```

Comment:
I think we can only ignore PVs in Pending.

We need to keep matching PVs that are also Bound.  Because we may still have to finish the PVC->PV binding half.

Reply:
Thanks, makes sense to me.  Can you add some unit tests to check that we don't choose PVs in non-Available states?

Reply:
Added in https://github.com/kubernetes/kubernetes/pull/68673/commits/1f8b4b1e1444529d6a536180d48c775e0aac7109.